chaosreliably/controls/experiment.py

```diff
@@ -0,0 +1,310 @@
+from typing import Any, cast
+
+from chaoslib.types import Configuration, Experiment, Secrets
+from logzero import logger
+
+from chaosreliably import get_session
+from chaosreliably.types import (
+    EntityContext,
+    EntityContextExperimentEventLabels,
+    EntityContextExperimentLabels,
+    EntityContextExperimentRunLabels,
+    EntityContextExperimentVersionLabels,
+    EntityContextMetadata,
+    EventType,
+)
+
+__all__ = ["before_experiment_control"]
+
+
+def before_experiment_control(
+    context: Experiment,
+    configuration: Configuration = None,
+    secrets: Secrets = None,
+    **kwargs: Any,
+) -> None:
+    """
+    Control run *before* the execution of an Experiment.
+
+    For a given Experiment, the control creates (if not already created) an Experiment
+    Entity Context and an Experiment Version Entity Context in the Reliably Entity
+    Server.
+
+    A unique Experiment Run Entity Context is also created, with an Experiment Event
+    Entity Context of type `EXPERIMENT_START` created, relating to the run.
+
+    The control requires the `arguments` of `commit_hash`, `source`, and `user` to be
+    provided to the control definition. If not provided, the control will simply not
+    create any Entity Contexts.
+
+    Once the Entity Contexts have been created, an entry into the configuration is made
+    under configuration["chaosreliably"]["experiment_run_labels"] to allow for
```

Comment:
why not just `reliably`? `chaosreliably` is the name of the extension but the configuration relates to the service. `chaosreliably` isn't really a thing.

Reply:
I guess my reasoning here is that it's not configuration specific to `reliably` as the service, it's configuration specific to the extension, nothing in `configuration["chaosreliably"]` is specific to us interacting with reliably itself.. It might also be easier to identify it as belonging to the extension?

I'm honestly not sure on this one.